src/System.Private.ServiceModel/src/System/ServiceModel/Channels/SocketConnection.cs

```diff
@@ -662,12 +724,12 @@ private void TraceWriteStart(int size, bool async)
             catch (SocketException socketException)
             {
                 throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(
-                    ConvertSendException(socketException, TimeSpan.MaxValue));
+                    ConvertSendException(socketException, TimeSpan.MaxValue, _asyncSendTimeout));
             }
             catch (ObjectDisposedException objectDisposedException)
             {
                 Exception exceptionToThrow = ConvertObjectDisposedException(objectDisposedException, TransferOperation.Write);
-                if (ReferenceEquals(exceptionToThrow, objectDisposedException))
+                if (object.ReferenceEquals(exceptionToThrow, objectDisposedException))
                 {
```

Comment:
I think there should be a VS Roslyn suggested fix here to simplify the code to remove object. as it's unnecessary. If there is, choose to apply the fix to the entire document.

Reply:
👍